### DISCUSSION TITLE
FOSDEM devroom-style use cases

### DISCUSSION THREAD

**Discussion by RichiH** [pretalx/pretalx#1220] [Ideas]
FOSDEM is considering to replace Pentabarf (shocking, I know!) for the 2023 edition. One of the requirements we would have is to be able to designate devroom managers who own and run CfPs for time slots which we assigned to them. I.e.:

* FOSDEM opens CfP for main track; accepts, assigns to rooms, rejects
* FOSDEM opens call for project booths; accepts, assigns to locations, rejects
* FOSDEM opens call for devrooms; accepts, assigns to rooms, rejects

Main track & projects booths are the trivial case.

For the accepted devrooms, they are basically small conferences happening at a time & place we give them. They would roughly:

* Open CfP on their own; accept, assign to time slot within their room, reject
* Sometimes move talks to different devrooms, potentially with the help of FOSDEM staff
* Without seeing anything from the other devrooms

From what I can see, Pretalx does not currently support this use case. Would you be willing to implement it? If this would be dependent on FOSDEM using Pretalx, we could come up with a checklist before commitments are made on either side.

**Comment by toshywoshy** [comment]
At present Pretalx can have different teams, each team has team admins.
In FOSDEM terms, a devroom is a team and runs it's own CfP, each FOSDEM main organizer, a Pretalx admin has rights to all teams.
Within teams you can assign different rights and Pretalx admins rights to all team (devrooms).

IMHO, there are some other problems that need to be looked at also.
* FOSDEM needs an output plugin to push the schedule into the current static website, and scheduling flow.
* It will also need some kind of link to the video system. In previous conversations, Pretalx doesn't want to add video storage to Pretalx and I tend to agree with that view personally. Rixx created a vimeo plugin, I created a Youtube plugin, and I think FOSDEM will need one for the FOSDEM video system.
* Another problem is having a whole overview of the entire conference, as you would be able to check each individual team, however interactions between teams, lets say moving a talk to another team, and a master team overviews is not currently in Pretalx.

**Reply by yoe** [reply]
> IMHO, there are some other problems that need to be looked at also.
> 
> * FOSDEM needs an output plugin to push the schedule into the current static website, and scheduling flow.

Yes please. Preferably this would be in pentabarf XML (or something sufficiently close to it), because most of our other infrastructure expects that file format.

> * It will also need some kind of link to the video system. In previous conversations, Pretalx doesn't want to add video storage to Pretalx and I tend to agree with that view personally. Rixx created a vimeo plugin, I created a Youtube plugin, and I think FOSDEM will need one for the FOSDEM video system.

Video is handled through different systems for FOSDEM, and we do not anticipate changing those. As soon as there is a pentabarf XML output "somewhere", our video tools can handle them and do the things that are needed.

> * Another problem is having a whole overview of the entire conference, as you would be able to check each individual team, however interactions between teams, lets say moving a talk to another team, and a master team overviews is not currently in Pretalx.

This would indeed be quite crucial.

**Reply by RichiH** [reply]
If FOSDEM staff needed to move the talks between devrooms at the suggestion of devroom managers, that might be a feasible workaround. That being said, a way for devroom managers to suggest a target devroom directly within pretalx would be much better, of course.

As an aside, please see https://inclusivenaming.org/word-lists/tier-1/#master

**Reply by toshywoshy** [reply]
> > IMHO, there are some other problems that need to be looked at also.
> > 
> > * FOSDEM needs an output plugin to push the schedule into the current static website, and scheduling flow.
> 
> Yes please. Preferably this would be in pentabarf XML (or something sufficiently close to it), because most of our other infrastructure expects that file format.

At present I know that the schedule can be exported in 'frab' XML, I am not sure if that would work for FOSDEM pentabarf XML import.
Is there any documentation on the pentabarf XML format, so we can look at creating an output plugin that outputs FOSDEM pentbarf XML.


> 
> > * It will also need some kind of link to the video system. In previous conversations, Pretalx doesn't want to add video storage to Pretalx and I tend to agree with that view personally. Rixx created a vimeo plugin, I created a Youtube plugin, and I think FOSDEM will need one for the FOSDEM video system.
> 
> Video is handled through different systems for FOSDEM, and we do not anticipate changing those. As soon as there is a pentabarf XML output "somewhere", our video tools can handle them and do the things that are needed.

I wasn't suggesting to change the video system, however currently as a DevRoom manager, we do get links from pentabarf to the video system for approval, I would expect that we need a similar field in Pretalx.
I am not sure how FOSDEM populates these links currently in pentabarf, but as a DevRoom manager I would expect a similar flow for Pretalx, that links to the FOSDEM video system for approval.


> 
> > * Another problem is having a whole overview of the entire conference, as you would be able to check each individual team, however interactions between teams, lets say moving a talk to another team, and a *main* team overviews is not currently in Pretalx.
> 
> This would indeed be quite crucial.

We had this problem when running CfgMgmtCamp in the past, right after FOSDEM.
People summit to a specific team CfP, but we think it fits better in another team.

**Reply by toshywoshy** [reply]
> If FOSDEM staff needed to move the talks between devrooms at the suggestion of devroom managers, that might be a feasible workaround. That being said, a way for devroom managers to suggest a target devroom directly within pretalx would be much better, of course.

At present I know that reviewers can be limited to specific tracks, and so tracks can be DevRooms, however for scheduling there is currently no way to have this fine grade limitations.
